pkg/controllers/jobset_controller.go

```diff
@@ -384,14 +384,23 @@ func (r *JobSetReconciler) constructJob(js *jobset.JobSet, rjob *jobset.Replicat
 		Spec: *rjob.Template.Spec.DeepCopy(),
 	}
 
-	// Add restart-attempt count label, it should be equal to jobSet restarts
-	// to indicate is part of the current jobSet run.
+	// Add replica count as label to the job.
```

Comment:
lets add unit tests for `constructJobsFromTemplate`; to make it easy to test, we can remove it from being a member function of `JobSetReconciler`